src/main/java/attendance/domain/AttendanceBook.java

```diff
@@ -0,0 +1,53 @@
+package attendance.domain;
+
+import static attendance.constant.ErrorMessage.UNREGISTERED_NICKNAME;
+
+import java.time.LocalDateTime;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+
+public class AttendanceBook {
+    private final Map<String, Attendances> crewAttendances;
+
+    public AttendanceBook(final Map<String, Attendances> crewAttendances) {
+        this.crewAttendances = crewAttendances;
+    }
+
+    public void attend(final String nickname, final Attendance attendance) {
+        validateNickname(nickname);
+        Attendances attendances = crewAttendances.get(nickname);
+        attendances.add(attendance);
+        crewAttendances.put(nickname, attendances);
+    }
+
+    public Attendance updateAttendance(final String nickname, final LocalDateTime dateTime) {
+        Attendances attendances = crewAttendances.get(nickname);
+        Attendance attendance = attendances.updateAttendance(dateTime);
+        crewAttendances.put(nickname, attendances);
+        return attendance;
+    }
+
+    public Attendances findByNickname(final String nickname) {
+        validateNickname(nickname);
+        return crewAttendances.get(nickname);
+    }
+
+    public Attendance findByNicknameAndDate(final String nickname, final LocalDateTime dateTime) {
+        Attendances attendances = crewAttendances.get(nickname);
+        return attendances.findByDate(dateTime.toLocalDate());
+    }
+
+    public Map<String, Attendances> findPenaltyCrews() {
+        return crewAttendances.entrySet()
+                .stream()
+                .filter(entry -> !AttendancePenalty.NONE.equals(entry.getValue().calculatePenalty()))
+                .collect(Collectors.toMap(Entry::getKey, Entry::getValue));
+    }
+
+    private void validateNickname(final String nickname) {
+        if (!crewAttendances.containsKey(nickname)) {
+            throw new IllegalArgumentException(UNREGISTERED_NICKNAME.getMessage());
+        }
```

Comment:
```suggestion
        if (crewAttendances.containsKey(nickname)) {
            return;
        }
        throw new IllegalArgumentException(UNREGISTERED_NICKNAME.getMessage());
        }
```
!를 이용한 반대 로직은 이해하기 어렵고 실수하기 쉬워 지양하는 편이에요~!

Reply:
부정 연산자도 다른 개발자가 봤을 때 그런 견해가 있을 수도 있겠군요!
다른 사람이 코드를 봤을 때를 더 고려해주어야겠어요

❓혹시, 부정 연산자를 조건문 내에서만 지양하시는 편인가요?
해당 피드백을 반영하며 이번에 리팩토링을 하다보니, controller의 executeMenu() 내에서
```java
return !Menu.QUIT.equals(selectedMenu);
```
이렇게 구현을 하게 되더라고요.
샐리는 혹시 반환값에 대해서도 부정연산자를 지양하시는 편인가요? 아니면 조건문 내에서만 지양하시는 편인가요?

Reply:
부정연산자는 개발자로 하여금 한번 더 반대로 생각하도록 하는 코드라 모든 코드에서 지양하는 편이에요! 사용하지 않고도 구현할 수 있거든요.

`return !Menu.QUIT.equals(selectedMenu);` -> QUIT과 동일하지 않은 값
`return Menu.QUIT.equals(selectedMenu);` -> QUIT